### DISCUSSION TITLE
Empty Chrome window always opens

### DISCUSSION THREAD

**Discussion by bas-ie** [cypress-io/cypress#14620] [Questions and Help]
Arch Linux
Cypress 6.1.0 (but have seen this behaviour with many previous versions)
Node 14.15.4

Chrome is my default system browser. Whenever I do a Cypress test run, no matter what the browser settings (including the default Electron headless and `--browser chrome --headless`) an empty Chrome window pops open that must be manually closed. I've always just sort of put up with this in the past, but I'm curious: is this an expected behaviour, and if so, why?

**Comment by portyu9** [comment]
Hi @bas-ie 

This is not expected. On Linux, Cypress detects installed browsers at startup by running commands such as:

```bash
google-chrome --version
chrome --version
google-chrome-stable --version
```

If one of those Chrome launchers/wrappers ignores `--version`, it can open an empty Chrome window even when Cypress is actually running Electron or headless Chrome.

Run those commands manually. Whichever one opens the blank window is the problem.

Fix or remove that launcher/symlink from your `PATH` so that `--version` prints the browser version and exits.

You can confirm what Cypress is detecting with:

```bash
DEBUG=cypress:launcher* npx cypress run --browser electron
```

So the fix is at the Chrome executable/wrapper level, not the Cypress browser/headless setting.

**Reply by portyu9** [reply]
@bas-ie Please mark this answer as accepted if it helped, thank you.
